components/sup/src/cli.rs

```diff
@@ -0,0 +1,120 @@
+use clap::App;
+
+use hab::cli::{
+    sub_sup_bash, sub_sup_depart, sub_sup_run, sub_sup_secret, sub_sup_sh, sub_sup_term,
+    sub_svc_status,
+};
+use VERSION;
+
+pub fn cli<'a, 'b>() -> App<'a, 'b> {
+    clap_app!(("hab-sup") =>
+        (about: "The Habitat Supervisor")
+        (version: VERSION)
+        (author: "\nAuthors: The Habitat Maintainers <humans@habitat.sh>\n")
+        // set custom usage string, otherwise the binary
+        // is displayed confusingly as `hab-sup`
+        // see: https://github.com/kbknapp/clap-rs/blob/2724ec5399c500b12a1a24d356f4090f4816f5e2/src/app/mod.rs#L373-L394
+        (usage: "hab sup <SUBCOMMAND>")
+        (@setting VersionlessSubcommands)
+        (@setting SubcommandRequiredElseHelp)
+        // this is the _full_ list of supervisor related cmds
+        // they are all enumerated here so that the entire help menu
+        // can be displayed from `hab sup --help`
+        (subcommand: sub_sup_bash().aliases(&["b", "ba", "bas"]))
+        (subcommand: sub_sup_depart().aliases(&["d", "de", "dep", "depa", "depart"]))
+        (subcommand: sub_sup_run().aliases(&["r", "ru"]))
+        (subcommand: sub_sup_secret().aliases(&["sec", "secr"]))
+        (subcommand: sub_sup_sh().aliases(&[]))
+        (subcommand: sub_svc_status().aliases(&["stat", "statu"]))
+        (subcommand: sub_sup_term().aliases(&["ter"]))
+    )
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
I agree with your thinking here, but I think this might be worth it's own PR. The way the CLI is defined is kinda weird and I think it deserves a bit more thought to unravel it all.

Reply:
One other benefit of this alternate approach is that you could leave this `cli` method where it is and avoid creating this new file.